ert/data/_record.py

```diff
@@ -249,6 +283,25 @@ async def transmit_file(
         pass
 
 
+# The use of this, rather ugly, function could possibly be avoided by moving some
+# of the functionality of the transmitters into the Record classes.
+def _make_record(
+    data: Any, index: Optional[RecordIndex] = None
+) -> Union[NumericalRecord, BlobRecord]:
+    validation_errors = []
+    try:
+        return NumericalRecord(data=data, index=index)
+    except ValidationError as err:
+        validation_errors.append(err)
+    try:
+        return BlobRecord(data=data)
+    except ValidationError as err:
+        validation_errors.append(err)
+    # This function only fails with multiple validation errors, not sure how to
+    # raise those.
+    raise Exception(validation_errors)
```

Comment:
I think the approach is fine of appending the messages. However, I would prefer if we raised something else then a bare exception. Perhaps a ValueError?

Reply:
ValueError is fine, done.